go-controller/pkg/util/slice.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+func remove[T comparable](slice []T, candidate T) []T {
+	idxsToRemove := make([]int, 0)
+	for i, sliceItem := range slice {
+		if sliceItem == candidate {
+			idxsToRemove = append(idxsToRemove, i)
+		}
+	}
+	for _, idxToRemove := range idxsToRemove {
+		slice = removeIndex(slice, idxToRemove)
+	}
+	return slice
```

Comment:
I would just export the two functions. Also I would explicitily doc them advertising they are not stable and modifies order.

```
func RemoveIndexFromSliceUnstable[T comparable](slice []T, i int) []T {
	var t T
	len := len(slice)
	slice[i] = slice[len-1]
        slice[len-1]=t                   // zero out the copied last element to have it garbage collected as early as possible
	return slice[:len-1]
}

func RemoveItemFromSliceUnstable[T comparable](slice []T, candidate T) []T {
	for i = 0; i < len(slice); {
		if slice[i] == candidate {
			slice = RemoveIndexFromSlice(slice,i)
                        continue
		}
                i++
	}
	return slice
}
```

https://go.dev/play/p/AJOBGQS2mPE

Reply:
Testing coverage would be nice for this one too

Reply:
That is sweet. Added TCs.
I didnt think about GC'ing any potential objects removed.

Reply:
1.21 is out of reach for now but in the future, we should use it for sure.

Reply:
..and by out of reach, i meant downstream, we arent using it. 

I renamed var `len` cause it clashed with a built-in func and added some simple input checks.